operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/ControllerConfiguration.java

```diff
@@ -51,15 +56,28 @@
   String labelSelector() default Constants.NO_VALUE_SET;
 
   /**
+   * Use onAddFilter, onUpdateFilter, onDeleteFilter instead.
+   *
    * <p>
    * Resource event filters only applies on events of the main custom resource. Not on events from
    * other event sources nor the periodic events.
    * </p>
    *
    * @return the list of event filters.
    */
+  @Deprecated
   Class<? extends ResourceEventFilter>[] eventFilters() default {};
 
+  /**
+   * Filter of onAdd events of resources. (Note that onDelete is missing, delete events don't
+   * trigger reconciliation - since if finalizer used there the event is an update that triggers the
+   * cleanup. If not used the resources are cleaned up by garbage collector.)
+   **/
+  Class<? extends Predicate<? extends HasMetadata>> onAddFilter() default VoidOnAddFilter.class;
```

Comment:
I don't have strong opinion on this. The only thing that is not nice on that is we need to implement our own operations for `add` and `or`. In addition to that this is not one interface, then it is three interfaces for every operation, because those have different parameters. And note that we are already passing all the input from the upstream api (informer, even source), so not sure if that API is somewhere to evolve.